server/src/main/java/org/opensearch/extensions/rest/RestInitializeExtensionAction.java

```diff
@@ -0,0 +1,122 @@
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.opensearch.rest.RestRequest.Method.POST;
+
+/**
+ * An action that initializes an extension
+ */
+public class RestInitializeExtensionAction extends BaseRestHandler {
+
+    private final ExtensionsManager extensionsManager;
+
+    @Override
+    public String getName() {
+        return ExtensionsManager.REQUEST_EXTENSION_ACTION_NAME;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/_extensions/initialize"));
+    }
+
+    public RestInitializeExtensionAction(ExtensionsManager extensionsManager) {
+        this.extensionsManager = extensionsManager;
+    }
+
+    @Override
+    public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        String name = "";
+        String uniqueId = "";
+        String hostAddress = "";
+        String port = "";
+        String version = "";
+        String openSearchVersion = "";
+        String minimumCompatibleVersion = "";
+        List<ExtensionDependency> dependencies = new ArrayList<>();
+
+        if (request.hasContent()) {
+            try (XContentParser parser = request.contentParser()) {
+                parser.nextToken();
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                    String currentFieldName = parser.currentName();
+                    parser.nextToken();
+                    if ("name".equals(currentFieldName)) {
+                        name = parser.text();
+                    } else if ("uniqueId".equals(currentFieldName)) {
+                        uniqueId = parser.text();
+                    } else if ("hostAddress".equals(currentFieldName)) {
+                        hostAddress = parser.text();
+                    } else if ("port".equals(currentFieldName)) {
+                        port = parser.text();
+                    } else if ("version".equals(currentFieldName)) {
+                        version = parser.text();
+                    } else if ("opensearchVersion".equals(currentFieldName)) {
+                        openSearchVersion = parser.text();
+                    } else if ("minimumCompatibleVersion".equals(currentFieldName)) {
+                        minimumCompatibleVersion = parser.text();
+                    } else if ("dependencies".equals(currentFieldName)) {
+                        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                            dependencies.add(ExtensionDependency.parse(parser));
```

Comment:
Make sure whatever exception you throw from parsing this (as suggested above) is handled here.